src/battle_util.c

```diff
@@ -11250,11 +11283,18 @@ bool8 AreBattlersOfOppositeGender(u8 battler1, u8 battler2)
     return (gender1 != MON_GENDERLESS && gender2 != MON_GENDERLESS && gender1 != gender2);
 }
 
-u32 CalcSecondaryEffectChance(u8 battler, u8 secondaryEffectChance)
+u32 CalcSecondaryEffectChance(u8 battler, u16 move)
 {
-    if (GetBattlerAbility(battler) == ABILITY_SERENE_GRACE)
+    u8 secondaryEffectChance = gBattleMoves[move].secondaryEffectChance;
+    bool8 hasSereneGrace = (GetBattlerAbility(battler) == ABILITY_SERENE_GRACE);
+
+    if (gSideStatuses[GetBattlerSide(battler)] & SIDE_STATUS_RAINBOW && hasSereneGrace && gBattleMoves[move].effect == EFFECT_FLINCH_HIT)
+        return secondaryEffectChance;
```

Comment:
As we discussed earlier, the chance for flinching is still x2 for either, it's just not x4 when both are active.

Reply:
Yep, I missed this one.